joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -525,6 +540,28 @@ CallbackReturn JointTrajectoryController::on_configure(const rclcpp_lifecycle::S
     return CallbackReturn::FAILURE;
   }
 
+  // TODO(livanov93): change when other option is implemented
+  if (has_velocity_command_interface_ && use_closed_loop_pid_adapter)
+  {
+    size_t num_joints = joint_names_.size();
+    pids_.resize(num_joints);
+    velocity_ff_.resize(num_joints);
+    command_.resize(num_joints, 0.0);
+
+    // Init PID gains from ROS parameter server
+    for (size_t i = 0; i < pids_.size(); ++i)
+    {
+      const std::string prefix = "gains." + joint_names_[i];
+      const auto k_p = auto_declare<double>(prefix + ".p", 0.0);
+      const auto k_i = auto_declare<double>(prefix + ".i", 0.0);
+      const auto k_d = auto_declare<double>(prefix + ".d", 0.0);
+      const auto i_clamp = auto_declare<double>(prefix + ".i_clamp", 0.0);
+      velocity_ff_[i] = auto_declare<double>("velocity_ff/" + joint_names_[i], 0.0);
```

Comment:
Propose also to rename the parameter

```suggestion
      ff_velocity_scale_[i] = auto_declare<double>("ff_velocity_scale/" + joint_names_[i], 0.0);
```

Reply:
ros-controls/ros2_controllers@09e6954